tests/docker/Dockerfile

```diff
@@ -79,21 +79,21 @@ RUN mkdir -p "/opt/kafka-2.3.1" && chmod a+rw /opt/kafka-2.3.1 && curl -s "$KAFK
     ldconfig
 
 # Install dependencies for kafka clients such as sarama and franz-go
-RUN git -C /opt clone https://github.com/Shopify/sarama.git && \
+RUN git -C /opt clone -b v1.32.0 --single-branch https://github.com/Shopify/sarama.git && \
     cd /opt/sarama/examples/interceptors && go mod tidy && go build && \
     cd /opt/sarama/examples/http_server && go mod tidy && go build && \
     cd /opt/sarama/examples/consumergroup && go mod tidy && go build && \
     cd /opt/sarama/examples/sasl_scram_client && go mod tidy && go build && \
-    git -C /opt clone https://github.com/twmb/franz-go.git && cd /opt/franz-go && \
-    cd /opt/franz-go/examples/bench && go mod tidy && go build && \
+    git -C /opt clone -b v1.5.0 --single-branch https://github.com/twmb/franz-go.git && \
+    cd /opt/franz-go && cd /opt/franz-go/examples/bench && go mod tidy && go build && \
     git -C /opt clone --branch ducktape2 https://github.com/vectorizedio/kafka-streams-examples.git && \
     cd /opt/kafka-streams-examples && mvn -DskipTests=true clean package
 
 # Install our in-tree go tests
 COPY --chown=0:0 tests/go /tmp/go
 RUN cd /tmp/go/sarama/produce_test && go mod tidy && go build
 
-RUN go install github.com/twmb/kcl@latest && \
+RUN go install github.com/twmb/kcl@v0.8.0 && \
```

Comment:
in the spirit of pinning golang deps, this seems sensible, but it is missing more code changes as it is part of a bigger changeset: PR #4119

i think only bringing in the change to pin to v0.8.0 is fine to keep because it protects from breaking changes on [master](https://github.com/twmb/kcl/tree/master) branch of `github.com/twmb/kcl` in the future.

@jcsp or @twmb let us know if the other parts of PR #4119 should also be backported